### PR TITLE
Add MODULE.bazel to Starlark language

### PR DESCRIPTION
**Pull request by jsharpe**
## Description

Adds filenames that are used by the new bzlmod format introduced in bazel 5.0.0 as an experimental feature and released in bazel 6.0.0.
Also added copy.bara.sky which is the main configuration file for Google's copybara tool.

## Checklist:

- [x] **I am associating a language with a new file extension.**
  - [x] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      
      - https://github.com/search?q=path%3A**%2FMODULE.bazel&type=code
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/bazelbuild/bazel/blob/c1168e13965ccd61ae9d948cf1e30bcdf459ae6a/MODULE.bazel
    - Sample license(s): APL 2.0
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

**Comment by lildude** [comment]
Of these, only `MODULE.bazel` has sufficient popularity on GitHub for inclusion right now. The others are a long way off our usage requirements.

**Comment by lildude** [comment]
Thinking about it further, there's no need for the `MODULE.bazel` filename either as it's already detected correctly because of the extension.

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -6491,9 +6491,12 @@ Starlark:
   - BUCK
   - BUILD
   - BUILD.bazel
+  - copy.bara.sky
```
Comment (jsharpe) [inline lib/linguist/languages.yml:6494]:
```suggestion
```

File: lib/linguist/languages.yml [hunk 2]
```diff
@@ -6491,9 +6491,12 @@ Starlark:
...
+  - MODULE.bazel
   - Tiltfile
   - WORKSPACE
   - WORKSPACE.bazel
+  - WORKSPACE.bzlmod
```
Comment (jsharpe) [inline lib/linguist/languages.yml:6499]:
```suggestion
```

**Comment by jsharpe** [comment]
> Of these, only `MODULE.bazel` has sufficient popularity on GitHub for inclusion right now. The others are a long way off our usage requirements.

Have update the PR to remove the ones that don't meet usage requirements. `WORKSPACE.bzlmod` is just a file that is used during transition to bzlmod so its usage will be expected to be low as bzlmod is adopted.

**Comment by jsharpe** [comment]
Do you need me to add a sample MODULE.bazel file?

**Comment by lildude** [comment]
> Do you need me to add a sample MODULE.bazel file?

Yes please.

**Comment by lildude** [comment]
> Yes please.

NM. I've added one directly to your branch.

**Comment by jsharpe** [comment]
> > Yes please.
> 
> NM. I've added one directly to your branch.

Thanks! Apologies for the lack of response sooner.

**Review by lildude — APPROVED** [review]

**Pull request merged**